File: src/main/java/com/syntomic/bigdata/AbstractJob.java
package com.syntomic.bigdata;

import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;

/** 作业基类 */
public abstract class AbstractJob {

    protected Configuration conf;
    protected StreamExecutionEnvironment env;
    protected StreamTableEnvironment tableEnv;

    public abstract void run() throws Exception;

    public void setConf(Configuration conf) {
        this.conf = conf;
    }

    public void setEnv(StreamExecutionEnvironment env) {
        this.env = env;
    }

    public void setTableEnv(StreamTableEnvironment tableEnv) {
        this.tableEnv = tableEnv;
    }
}

File: src/main/python/com/syntomic/bigdata/abstract_job.py
from abc import ABC
from typing import Dict

from pyflink.datastream import StreamExecutionEnvironment
from pyflink.table import StreamTableEnvironment

class AbstractJob(ABC):

    def __init__(self, conf: Dict[str, str]):
        self.conf = conf
        self.env = StreamExecutionEnvironment.get_execution_environment()
        self.env.get_config().set_global_job_parameters(conf)
        self.tableEnv = StreamTableEnvironment.create(stream_execution_environment=self.env)

    def run(self):
        pass